### PR TITLE
Getting Started With Node.js Timers Tutorial: Everything you need to know 

### LINKED ISSUE #2048: Getting Started With Node.js Timers: Everything you need to know
### Proposed title of the article
Getting Started With Node.js Timers: Everything you need to know

### Introduction paragraph (2-3 paragraphs):
The Timers module in Node.js contains various functions that allow us to execute a block of code or a function after a set period of time. The Timers module is global, we do not need to use require() to import it.
Timers module has the following functions:

Scheduling Timers: It is used to call a function after a set period of time.
setImmediate()
setInterval()
setTimeout()
Cancelling Timers: It is used to cancel the scheduled timer.
clearImmediate()
clearInterval()
clearTimeout()

1. setImmediate() method: It schedules the “immediate” execution of the callback after I/O events callbacks. In the following example, multiple setImmediate functions are called. Whenever we do these callback functions are queued for execution in the order in which they are created. The entire callback queue is processed after every event loop iteration. If an immediate timer is queued from inside an executing callback, that timer will not be triggered until the next event loop iteration.  
`js
setImmediate(function A() {
    setImmediate(function B() {
      console.log(1);
      setImmediate(function D() { 
        console.log(2);
      });
    });
  
    setImmediate(function C() {
      console.log(3);
      setImmediate(function E() { 
        console.log(4);
      });
    });
});
...................................
  
console.log('Started...');
`

### Key takeaways: 

- Scheduling Timers: It is used to call a function after a set period of time.
setImmediate()
setInterval()
setTimeout()
- Cancelling Timers: It is used to cancel the scheduled timer.
clearImmediate()
clearInterval()
clearTimeout()

### PR DESCRIPTION
**Pull request by owinowendy**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #2048

### CODE REVIEW

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @owinowendy,

Great article.

I've suggested few changes. Please let me know when it's done, we can move on to the tech review.

Thank you.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by srishilesh** [comment]
@owinowendy 

I'm so sorry.

Due to certain personal emergencies, I was not able to review them.

So, I'm assigning it to @cyberShaw . He'll continue the review from here.

Thank you.

**Comment by cyberShaw** [comment]
I'll do the review by today, @owinowendy!

File: articles/nodejs-timers/index.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+**Output**:
+
+```bash
+before a set immediate function is called
+after the immediate function has been executed
+executing the immediate function: undefined
+```
+
+While executing this method, you're likely to encounter an error as seen below, otherwise, skip this part:
```
Comment (cyberShaw) [inline articles/nodejs-timers/index.md:136]: Please do explain why this error is likely to be encountered in a few lines.
And please also explain why the fix you suggested works that way. Readers would get a deeper understanding of why `setImmediate()` breaks and throws an error in some cases.

**Review by cyberShaw — CHANGES_REQUESTED** [review]
Great article, @owinowendy!
Please make the suggested changes and we can move to final review.

**Comment by cyberShaw** [comment]
This article is ready for final review. Great job, @owinowendy 

cc: @hectorkambow 🚀

**Review by cyberShaw — APPROVED** [review]

**Comment by hectorkambow** [comment]
@owinowendy 

just need to preview and published - when repo changes are finalized

NOTE: Thank you - we are making some changes to the main repo atm - and this is delaying a few article from being published as we want to implement the changes 1st - once these are done - we will publish this one asap. Thank you 🚀

**Comment by hectorkambow** [comment]
https://www.section.io/engineering-education/nodejs-timers/

published

**Pull request closed without merging**